build-tools/packages/build-cli/src/handlers/checkFunctions.ts

```diff
@@ -903,3 +903,54 @@ export const checkValidReleaseGroup: StateHandlerFunction = async (
 
 	return true;
 };
+
+/**
+ * Checks whether layer compat generation needs to be updated.
+ *
+ * @param state - The current state machine state.
+ * @param machine - The state machine.
+ * @param testMode - Set to true to run function in test mode.
+ * @param log - A logger that the function can use for logging.
+ * @param data - An object with handler-specific contextual data.
+ * @returns True if the state was handled; false otherwise.
+ */
+export const checkLayerCompatGeneration: StateHandlerFunction = async (
+	state: MachineState,
+	machine: Machine<unknown>,
+	testMode: boolean,
+	log: CommandLogger,
+	data: FluidReleaseStateHandlerData,
+): Promise<boolean> => {
+	if (testMode) return true;
+
+	const { context, bumpType } = data;
+	const gitRepo = await context.getGitRepository();
+
+	if (bumpType === "patch") {
+		log.verbose(`Skipping layer compat generation check for patch release.`);
+		BaseStateHandler.signalSuccess(machine, state);
+		return true;
+	}
+
+	// layerGeneration:gen should be run from the root. It will only update packages that have the layerGeneration:gen
+	// script defined in their package.json.
+	const result = await execa.command(`pnpm run -r layerGeneration:gen`, {
+		cwd: context.root,
+	});
+	log.verbose(result.stdout);
```

Comment:
Is there a reason we can't call the code that this command invokes? Exec-ing is the least preferred method of integrating if we can just call a function.

Reply:
I don't think we can call the code in generate commands which the exec invokes for a couple of reasons:
1. That code needs to be called for every package. Although I think there is a way to do that from here by getting all the packages for a releaseGroup.
2. The `UpdateGenerationCommand:processPackage` function expects a few flags and every package can pass in different values for them. I don't know if there is a way to get that infor from here and pass it.